File: src/main/java/siosio/doma/inspection/DaoInspectionToolProvider.java
package siosio.doma.inspection;

import com.intellij.codeInspection.InspectionToolProvider;

public class DaoInspectionToolProvider implements InspectionToolProvider {

    @Override
    public Class[] getInspectionClasses() {
        return new Class[]{DaoInspectionTool.class};
    }
}

File: src/main/java/siosio/doma/inspection/dao/DaoInspectionToolProvider.kt
package siosio.doma.inspection.dao

import com.intellij.codeInspection.*

class DaoInspectionToolProvider : InspectionToolProvider {
    override fun getInspectionClasses(): Array<Class<out LocalInspectionTool>> {
        return arrayOf(DaoInspectionTool::class.java, KotlinDaoInspectionTool::class.java)
    }
}
